pkg/kubectl/rollback.go

```diff
@@ -283,6 +283,44 @@ func (r *DaemonSetRollbacker) Rollback(obj runtime.Object, updatedAnnotations ma
 	return rollbackSuccess, nil
 }
 
+// match check if the given DaemonSet's template matches the template stored in the given history.
+// TODO: This is copied from pkg/controller/daemon, de-duplicate it after migrating from extensions to apps
+func match(ds *externalextensions.DaemonSet, history *appsv1beta1.ControllerRevision) (bool, error) {
```

Comment:
Another alternative is to expose `match` function.

Reply:
Expose `match` for what purpose?

Reply:
To reuse the function?

Reply:
Who's going to use this function other than kubectl?

Reply:
It is previously used by some function from `pkg/controller/daemon`, and now by kubectl. As it is used by more than one party, perhaps we can expose it (i.e., `Match`), or move the function to a utility directory?

Reply:
Look closer at https://github.com/kubernetes/kubernetes/pull/51157/commits/05028d320f4fe6f2b5042e14eb59723c3e41339d (Copy DaemonSet controller codes to kubectl).

This function was originally expose and shared already. We have to duplicate it because we can't use apps DaemonSet in kubectl yet, and kubectl and ds controller share the same codes. kubectl's `match` uses extensions DaemonSet, and daemonset controller's `Match` uses apps DaemonSet. 

Therefore we need to duplicate these functions, and have their input DaemonSet of different api versions.

Reply:
I was confused by the godoc explanation. Although this is a copied function, but it is technically speaking a different function due to different function parameter.

What i did to deployment is also duplicating the necessary functions, but renamed them differently based on the version. For example, in this case, `MatchExtensionsV1Beta2`. I think it is better in this case (obvious in such a way that we will not forget removing the function when retiring the old version). Just my 2 cent. :)

Reply:
Renaming it sounds like a good idea.

Reply:
Actually, I prefer not to rename functions, for the following reasons:

1. It's easier to search for the other duplicate in the future when someone wants to de-duplicate it, even if the code gets moved 
1. Shorter name is preferred 
1. It's impossible for callers to use the wrong function, because the parameter won't match 

We've included version in function names in our codebase before, but in those cases, functions live in the same place and can't have the same name.

Reply:
This sounds good too. I do not have strong opinion on this. The reason i commented was due to confusion by the godoc explanation. As both functions have different parameter, this will work too.